flink-java/src/main/java/org/apache/flink/api/java/operators/CrossOperator.java

```diff
@@ -129,14 +129,11 @@ private String getDefaultName() {
 
 		public DefaultCross(DataSet<I1> input1, DataSet<I2> input2, CrossHint hint, String defaultName) {
 			
-			super(input1, input2, new DefaultCrossFunction<I1, I2>(),
+			super(Preconditions.checkNotNull(input1, "input1 is null"),
```

Comment:
DefaultCross calls `input1.getType()` and `input2.getType()` before calling super() on line 134. So, if we add null check to super class (e.g. TwoInputOperator) it will not work for DefaultCross